### ISSUE TITLE
Error "ASLR detected: cannot continue" loading AsmProfiler.dll

### ISSUE THREAD

**Issue by CBertrand2** [andremussche/asmprofiler#7]
Hi
When my application tries to load AsmProfiler.dll, an exception occurs with this message: "ASLR detected: cannot continue".
Do you think it is possible to make this DLL be compatible with the Windows' ASLR protection ?
Of course, I cannot disable this protection.

**Comment by andremussche** [comment]
Yes, I think it is possible by getting the real HInstance of the application (instead of the default).
But this tool is mainly for profiling and debugging, so why can't you disable ASLR? (make a special build to do the investigation and optimization)

**Comment by CBertrand2** [comment]
Hi @andremussche 

> make a special build to do the investigation and optimization

In Delphi, compiling your project ? Could you tell how I can do that, please ?

**Comment by andremussche** [comment]
> > make a special build to do the investigation and optimization
>
> In Delphi, compiling your project ? Could you tell how I can do that, please ?

You can start with editing the debug configuration: 

- Project -> Options -> Building -> Linking:
  - Uncheck "Data Execution Prevention compatible" 
  - Uncheck "Support address space layout randomization (ASLR)" 

Or make a new configuration in the projects pane using RMB on "build configurations" -> "New configuration"
